### PR TITLE
pie-icons-vue@1.0.0-alpha.0 – Create PIE React Icons package

### PR DESCRIPTION
**Pull request by davidpn11**
### v1.17.0
_October 22, 2022_
--------------------------------------

### Added

- `pie-icons-react` package to `/tools`.
- add `/esm` folders to .gitignore

### v0.0.1
_October 12, 2022_
-----------------------------------

### Added

- Add scripts to generate the React icons
- Create build step using Rollup, bundling in CJS and ESM

### CODE REVIEW

File: packages/tools/pie-icons-react/scripts/react-icons-gen.js
Comment (ashleynolan) [inline packages/tools/pie-icons-react/scripts/react-icons-gen.js:9]: It would be good to keep the generated icon directory similar to the vue package.

We've put generated icons in a folder called `icons`, as we thought that using `src/` may be confusing (mainly as the files are generated). It also means that you can add the individual icon `tsx` files to the package bundle, so that applications could just choose to import an individual component file (if they don't use a build tool that allows them to tree-shake for instance).

Maybe can catch up later when you're online about this one?

File: packages/tools/pie-icons-react/package.json
Comment (ashleynolan) [inline packages/tools/pie-icons-react/package.json:8]: Would be good to include the individual icon files here too as part of the bundle, just for any consuming applications that can't tree-shake (just gives more options for how they can consume the package potentially).

**Review by ashleynolan — COMMENTED** [review]
Looks good – a few comments, mainly to make sure we work out how to align the Vue and React package structures a little, but then LGTM 😄

File: CHANGELOG.md
Comment (davidpn11) [reply CHANGELOG.md:6]: Oh, thats crazy! It was an auto formatting on my IDE that changed the whole file. Going to ever these changes, sorry about that

**Review by ashleynolan — APPROVED** [review]
Looks good to me. Will get someone else on the team to take a look at it on Monday too.

File: packages/tools/pie-icons-react/package.json
Comment (jamieomaguire) [inline packages/tools/pie-icons-react/package.json:49]: some of these dependencies might be able to be moved up to the project root `package.json` - I can imagine `typescript`, `rollup` and `@babel/node` being reused, for example :)

**Review by jamieomaguire — DISMISSED** [review]

**Review by kevinrodrigues — APPROVED** [review]

File: packages/tools/pie-icons-react/package.json
Comment (ashleynolan) [reply packages/tools/pie-icons-react/package.json:49]: Will look to do this as part of some tidy up PRs that will happen after we merge this if that's ok?

**Review by ashleynolan — APPROVED** [review]

**Pull request merged**
